### ISSUE TITLE
Port SSL endpoints to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6157]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#6157|
|Type|enhancement|
|Created|2012-10-23 14:35:34Z|
|Branch|https://github.com/twisted/twisted/tree/sslendpoints-py3-6157|

`twisted.internet.endpoints` SSL endpoints should be ported to Python 3 (they're required by the SSL unit tests, for example).

<details><summary>Searchable metadata</summary>

```
trac-id__6157 6157
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_sslendpoints_py3_6157 branches/sslendpoints-py3-6157
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1351002934000000 1351002934000000
changetime__1352225107000000 1352225107000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

(In [36351]) Branching to 'sslendpoints-py3-6157'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst set owner to @exarkun|
|-|-|

Ready for review; all I did was move code around.

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/sslendpoints-py3-6157

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @itamarst|
|-|-|

Thanks.  Looks pretty good.  There are some issues though, enumerated by the build results.  Fix those then merge.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set status to `closed`|
|-|-|

(In [36354]) Merge sslendpoints-py3-6157.

Author: itamar
Review: exarkun
Fixes: [#6157](https://github.com/twisted/twisted/issues/6157)

Port SSL endpoints to Python 3.
